code/modules/cargo/markets/market_telepad.dm

```diff
@@ -48,40 +56,274 @@
+		. += "[base_icon_state]_closed"
+	else
+		var/mutable_appearance/overlay = mutable_appearance(icon, "[base_icon_state]_open")
+		overlay.pixel_w -= 2
+		overlay.pixel_z -= 1
+		. += overlay
+
+/obj/machinery/ltsrbt/attack_hand(mob/user, list/modifiers)
+	. = ..()
+	if(.)
+		return
+	if(!state_open)
+		open_machine(density_to_set = TRUE)
+	else
+		close_machine()
+
+/obj/machinery/ltsrbt/open_machine(drop = TRUE, density_to_set = FALSE)
+	. = ..()
+	playsound(src, 'sound/machines/oven/oven_open.ogg', 75, TRUE)
+
+/obj/machinery/ltsrbt/close_machine(atom/movable/target, density_to_set = TRUE)
+	. = ..()
+	playsound(src, 'sound/machines/oven/oven_close.ogg', 75, TRUE)
+
+/obj/machinery/ltsrbt/set_occupant(obj/item/new_occupant)
+	. = ..()
+	if(new_occupant)
+		current_name = new_occupant.name
+		current_desc = new_occupant.desc
+
+/obj/machinery/ltsrbt/can_be_occupant(atom/movable/atom)
+	return isitem(atom) && !atom.anchored
+
+/obj/machinery/ltsrbt/Exited(atom/movable/gone)
+	if(gone == occupant)
+		current_price = initial(current_price)
+		current_name = ""
+		current_desc = ""
+		update_appearance(UPDATE_ICON_STATE)
+	return ..()
+
+/obj/machinery/ltsrbt/attack_hand_secondary(mob/user, list/modifiers)
+	. = ..()
+	if(. == SECONDARY_ATTACK_CANCEL_ATTACK_CHAIN)
+		return
+	if(state_open)
+		balloon_alert(user, "close it first!")
+		return SECONDARY_ATTACK_CANCEL_ATTACK_CHAIN
+	if(!occupant)
+		balloon_alert(user, "nothing loaded!")
+		return SECONDARY_ATTACK_CANCEL_ATTACK_CHAIN
+	if(machine_stat & NOPOWER)
+		balloon_alert(user, "machine unpowered!")
+		return SECONDARY_ATTACK_CANCEL_ATTACK_CHAIN
+	if(!COOLDOWN_FINISHED(src, recharge_cooldown))
+		balloon_alert(user, "on cooldown!")
+		return SECONDARY_ATTACK_CANCEL_ATTACK_CHAIN
+	ui_interact(user)
+	return SECONDARY_ATTACK_CANCEL_ATTACK_CHAIN
+
+/obj/machinery/ltsrbt/item_interaction(mob/living/user, obj/item/tool, list/modifiers)
+	if(user.combat_mode)
+		return NONE
+
+	var/creds_value = tool.get_item_credit_value()
+
+	if(state_open)
+		if(locate(/mob/living) in tool.get_all_contents())
+			say("Living being detected, cannot sell!")
+			playsound(src, 'sound/machines/buzz-sigh.ogg', 40, FALSE)
+			return ITEM_INTERACT_BLOCKING
+		if(!user.transferItemToLoc(tool, src))
+			balloon_alert(user, "stuck to your hands!")
+			return ITEM_INTERACT_BLOCKING
+		balloon_alert(user, "item loaded")
+		close_machine(tool)
+		return ITEM_INTERACT_SUCCESS
+	else if(!creds_value)
+		balloon_alert(user, "open the machine!")
+		return ITEM_INTERACT_BLOCKING
+
+	if(machine_stat & NOPOWER)
+		return
+
+	if(creds_value < restock_cost)
+		say("Insufficient credits!")
+		playsound(src, 'sound/machines/buzz-sigh.ogg', 40, FALSE)
+		return ITEM_INTERACT_BLOCKING
+
+	if(istype(tool, /obj/item/holochip))
+		var/obj/item/holochip/chip = tool
+		chip.spend(restock_cost)
+	else
+		qdel(tool)
+		if(creds_value != restock_cost)
+			var/obj/item/holochip/change = new(creds_value - restock_cost)
+			user.put_in_hands(change)
```

Comment:
In testing, this seems to produce a 1 holocredit chip when trying to restock the market with no-holocredit money sources. Tested via c10000 bills on the machine as well as with smaller bills.